talon.wdl

```diff
@@ -463,7 +463,10 @@ task Talon {
     command <<<
         set -e
         mkdir -p "$(dirname ~{outputPrefix})"
-        export TMPDIR=/tmp
+        mkdir -p $PWD/tmp #Standard /tmp fills up which makes the SQLite process crash.
+        ln -s $PWD/tmp /tmp/sqltmp #Multiprocessing will crash if the absolute path is too long.
```

Comment:
Softlinking in containers will not work.
Why not 
```bash
mkdir ./tmp
export TMPDIR=./tmp/
``` 
?
This will keep the path short and it should be inside the cromwell-executions directory.

Reply:
The new code you suggested sadly still results in the following error:
```
Process SyncManager-5:
Traceback (most recent call last):
  File "/usr/local/lib/python3.7/multiprocessing/process.py", line 297, in _bootstrap
    self.run()
  File "/usr/local/lib/python3.7/multiprocessing/process.py", line 99, in run
    self._target(*self._args, **self._kwargs)
  File "/usr/local/lib/python3.7/multiprocessing/managers.py", line 589, in _run_server
    server = cls._Server(registry, address, authkey, serializer)
  File "/usr/local/lib/python3.7/multiprocessing/managers.py", line 145, in __init__
    self.listener = Listener(address=address, backlog=16)
  File "/usr/local/lib/python3.7/multiprocessing/connection.py", line 438, in __init__
    self._listener = SocketListener(address, family, backlog)
  File "/usr/local/lib/python3.7/multiprocessing/connection.py", line 581, in __init__
    self._socket.bind(address)
OSError: AF_UNIX path too long
Traceback (most recent call last):
  File "/usr/local/bin/talon", line 8, in <module>
    sys.exit(main())
  File "/usr/local/lib/python3.7/site-packages/talon/talon.py", line 2385, in main
    manager = mp.Manager()
  File "/usr/local/lib/python3.7/multiprocessing/context.py", line 56, in Manager
    m.start()
  File "/usr/local/lib/python3.7/multiprocessing/managers.py", line 567, in start
    self._address = reader.recv()
  File "/usr/local/lib/python3.7/multiprocessing/connection.py", line 250, in recv
    buf = self._recv_bytes()
  File "/usr/local/lib/python3.7/multiprocessing/connection.py", line 407, in _recv_bytes
    buf = self._recv(4)
  File "/usr/local/lib/python3.7/multiprocessing/connection.py", line 383, in _recv
    raise EOFError
EOFError
```

We were able to prevent this error from occurring with the code suggestion in the PR. The soft linking did work in a actual test run on SHARK.

Reply:
Okay. I understand. Does this work locally in docker as well?

Reply:
Since the softlink is created inside the container and used only inside the container, it should be fine.